src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -368,7 +368,7 @@ void primitive_inst::update_shape() {
     if (_node->is_in_shape_of_subgraph()) {
         bool subgraph_input_changed = false;
         for (size_t i = 0; i < dependant_shape_of_insts.size(); i++) {
-            if (dependant_shape_of_insts[i]->get_flag(ExecutionFlags::SHAPE_CHANGED)) {
+            if (dependant_shape_of_insts[i]->get_flag(ExecutionFlags::SHAPE_CHANGED) || dependant_shape_of_insts[i]->_node->is_type<input_layout>()) {
```

Comment:
Can't we rely on the common logic with `SHAPE_CHANGED` flag here and in other places if we set the flag in `input_layout::set_data` ?

Reply:
@vladimir-paramuzov , thanks, moved to input_layout_impl